pkg/kubelet/volume/nfs/nfs_test.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nfs
+
+import (
+	"testing"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/volume"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/types"
+)
+
+func TestCanSupport(t *testing.T) {
+	plugMgr := volume.PluginMgr{}
+	plugMgr.InitPlugins(ProbeVolumePlugins(), &volume.FakeHost{"fake", nil})
+	plug, err := plugMgr.FindPluginByName("kubernetes.io/nfs")
+	if err != nil {
+		t.Errorf("Can't find the plugin by name")
+	}
+	if plug.Name() != "kubernetes.io/nfs" {
+		t.Errorf("Wrong name: %s", plug.Name())
+	}
+	if !plug.CanSupport(&api.Volume{VolumeSource: api.VolumeSource{NFS: &api.NFSVolumeSource{}}}) {
+		t.Errorf("Expected true")
+	}
+	if plug.CanSupport(&api.Volume{VolumeSource: api.VolumeSource{}}) {
+		t.Errorf("Expected false")
+	}
+}
+
+type fakeNFSMounter struct{}
```

Comment:
Is this ready for review again?

On Mon, Mar 9, 2015 at 6:58 AM, Chris Alfonso <notifications@github.com>
wrote:

> In pkg/kubelet/volume/nfs/nfs_test.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/4601#discussion_r26036673>
> :
>
> > +	plug, err := plugMgr.FindPluginByName("kubernetes.io/nfs")
> > +	if err != nil {
> > +		t.Errorf("Can't find the plugin by name")
> > +	}
> > +	if plug.Name() != "kubernetes.io/nfs" {
> > +		t.Errorf("Wrong name: %s", plug.Name())
> > +	}
> > +	if !plug.CanSupport(&api.Volume{VolumeSource: api.VolumeSource{NFS: &api.NFSVolumeSource{}}}) {
> > +		t.Errorf("Expected true")
> > +	}
> > +	if plug.CanSupport(&api.Volume{VolumeSource: api.VolumeSource{}}) {
> > +		t.Errorf("Expected false")
> > +	}
> > +}
> > +
> > +type fakeNFSMounter struct{}
>
> The nfsMounter is used from nfs_mount.go because of the server paramter
> needed. The fake mounter has to use the same interfaces, so that's the
> reason for this fakeNFSMounter.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/4601/files#r26036673>
> .
>